Event/GetResponseSessionUserEvent.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace FOS\UserBundle\Event;
+
+use FOS\UserBundle\Model\UserInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class GetResponseSessionUserEvent extends GetResponseUserEvent
+{
+    /**
+     * @var UserInterface
+     */
+    private $sessionUser;
+
+    /**
+     * GetResponseSessionUserEvent constructor.
+     *
+     * @param UserInterface $user
+     * @param UserInterface $sessionUser
+     * @param Request       $request
+     */
+    public function __construct(UserInterface $user = null, UserInterface $sessionUser = null, Request $request = null)
```

Comment:
the current user may not be a FOSUserBundle one, and you have nothing checking it in the controller, making the code unsafe.

and IMO, we should not include it here. If the listener wants to check this based on the logged-in user, they should use the token_storage service (and btw, they probably *also* need the authorization checker, as the login page must not be forbidden to logged in users being only remembered)